utils: update tools script to have a semi-automatic octopus merge

acomodi opened the pull request:
Signed-off-by: Alessandro Comodi <acomodi@antmicro.com>

This PR adds a tool to semi-automatically update the various tools by making an octopus merge of all the active WIP branches of the selected tool

litghost commented:
CI is red due to formatting?

litghost commented on utils/update_tools/Makefile:
Should probably support parallel build here (e.g. '-jN')

acomodi replied:
Done

acomodi commented:
@litghost FIxed formatting.

The idea of this tool is also to be used for Yosys. I have also updated the names of the branches in the VtR fork and rebased all of them in master. 
Now all the branches that have to be included in the `octopus merge` will have the `wip/` prefix.

The problem is that all the Travis CI builds have started for each of the branch and they could take a long time. Is there a way to cancel them?

acomodi commented:
@mithro FYI

mithro commented on utils/update_tools/update_tools.py:
```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
```
Add top level module docstring.

acomodi replied:
Done.
